Feature auth

ShehabMohsen opened the pull request:
- The landing page will be viewable depending on whether the user has been authenticated.
- Heavy code base refactoring from renaming component names and setting up routes

raynelfss reviewed (approved):
LGTM, I added the axios dependency which was missing and giving me errors. Also fixed a mistake in login/register page.

**Ps:** I actually don't know how the login contexts work so maybe go with your gut for that one :v

ShehabMohsen commented:
> LGTM, I added the axios dependency which was missing and giving me errors. Also fixed a mistake in login/register page.
> 
> **Ps:** I actually don't know how the login contexts work so maybe go with your gut for that one :v

Thank you for adding it. 

Don't worry about the context, I'll handle it hopefully later in the evening or tomorrow. Everything is looking good.
